features/step_definitions/operators.rb

```diff
@@ -160,3 +160,46 @@
   cb_name = 'operator_channel_name' unless cb_name
   cb[cb_name] = cluster_version('version').channel.split('-')[1]
 end
+
+Given /^I wait(?: up to (\d+) seconds)? for operator "([^"]*)" to become conditions available\/non-progressing\/non-degraded: "([^"]*)"$/ do | time, operator, conditions |
```

Comment:
Starting a new thread. Not that above is wrong. Just it is very long step name and provides redundant information. My suggestion is to make step definition:
```
Given /^operator #{QUOTED} becomes (\S+) (?: within #{NUMBER} seconds)$/ do | operator_name, conditions, timeout |
```

Then `conditions` will be split on slash `/` and when `available` is one of the words, then we should check whether condition `Available` is `"True"`. If we see `non-avalable`, then we check that condition `Available` is `"False"`. If `available` and `non-available` are both missing, then we don't care about condition `Available`.

Same thing with `progressing` and `non-progressing`, ditto `degraded`, `non-degraded`.

If another word is found, then we return an error that such condition does not exist.

It is very similar why we write in code `if my_variable { ... }` instead of `if my_variable == true { ... }`. It is about brewity and conventions.

So later you will call the step like:
```
Then operator "my-operator" becomes available/non-progressing/non-degraded within 130 seconds
```
or
```
Then operator "my-operator" becomes non-available/progressing/degraded within 130 seconds
```

Reply:
It's good idea. I used regular expression to restrict input conditions, used one placeholder "_" to indicate the unconditional input. 
```
reg = /(\_|available|non-available)\/(\_|progressing|non-progressing)\/(\_|degraded|non-degraded)/
```
And we can call the step like:
Given operator "kube-apiserver" becomes "_/progressing/_" within 100 seconds
Given operator "kube-apiserver" becomes "available/non-progressing/non-degraded" within 1200 seconds